# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/WasmFeatures.h
# Commit: 3e40bdd11ab5
# Full Hash: 3e40bdd11ab5aa13a583162bb8394a234465c3a2
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 6
# Description:
#   Bug 1727084 - Memory64 - Ion operations. r=yury
#   
#   Add memory64 generalizations for Ion for all operations (except bulk
#   memory, done in an earlier patch).
#   
# ==============================================================================

diff -r 44b0c0bed8a6 -r 3e40bdd11ab5 js/public/WasmFeatures.h
--- a/js/public/WasmFeatures.h	Mon Oct 18 09:58:15 2021 +0000
+++ b/js/public/WasmFeatures.h	Mon Oct 18 09:58:16 2021 +0000
@@ -139,7 +139,8 @@
   EXPERIMENTAL(/* capitalized name   */ Memory64,                             \
                /* lower case name    */ memory64,                             \
                /* compile predicate  */ WASM_MEMORY64_ENABLED,                \
-               /* compiler predicate */ BaselineAvailable(cx),                \
+               /* compiler predicate */ BaselineAvailable(cx) ||              \
+                  IonAvailable(cx),                                           \
                /* flag predicate     */ !IsFuzzingIon(cx) &&                  \
                    !IsFuzzingCranelift(cx),                                   \
                /* shell flag         */ "memory64",                           \